Framework/Algorithms/src/Qxy.cpp

```diff
@@ -381,10 +392,8 @@ Qxy::setUpOutputWorkspace(API::MatrixWorkspace_const_sptr inputWorkspace) {
   // Fill the X vectors in the output workspace
   for (int i = 0; i < bins - 1; ++i) {
     outputWorkspace->setBinEdges(i, axis);
-    for (int j = 0; j < bins - j; ++j) {
```

Comment:
What is this upper loop bound?? It looks weird, but if it is correct your change below cannot be right.